index.js

```diff
@@ -40,23 +46,34 @@ async function installDotnetTool(toolName, version) {
 }
 
 // Export functions for testing
-exports.dafnyURL = dafnyURL;
+exports.dafnyURLAndFullVersion = dafnyURLAndFullVersion;
 exports.getDistribution = getDistribution;
-exports.latestNightlyVersionFromDotnetToolSearch =
-  latestNightlyVersionFromDotnetToolSearch;
+exports.resolveNightlyVersionFromDotnetToolSearch =
+  resolveNightlyVersionFromDotnetToolSearch;
 
-async function dafnyURL(version, distribution) {
+async function dafnyURLAndFullVersion(version, distribution) {
   const versionPath = version.startsWith("nightly") ? "nightly" : `v${version}`;
-  if (version == "nightly-latest") {
-    version = await latestNightlyVersion();
+  var fullVersion;
+  if (version.startsWith("nightly")) {
+    fullVersion = await resolveNightlyVersion(version);
+    // Slice off the "3.11.0.50201-" from 3.11.0.50201-nightly-2023-02-13-14bc57f, for e.g.
+    version = fullVersion.slice(fullVersion.indexOf("-") + 1);
```

Comment:
Is there a difference in the version between the latest version and the latest nightly?

Reply:
The idea is to resolve to a version string like `4.3.0-nightly-blah-blah`, from which you can infer that you're dealing with at least Dafny 4.3.0 (technically we're using pre-release strings wrong because that really means "a pre-release for the upcoming 4.3.0 release" when we really mean "for the next release a.k.a. 4.4.0", but that's something to fix in a different scope.

Reply:
Yes, this is exactly what I'm saying. `4.3.0-nightly-blah-blah` is not really `4.3.0` it is something else.
But as I read more, this is not really in scope here. I'm not sure what version you would send through.
Since it would (should) eventually line up to things like the runtime version